code/game/machinery/doors/firedoor.dm

```diff
@@ -78,17 +87,158 @@
+				"<span class='notice'>You pry open \the [src] with your [hand_string]!</span>")
+
+			open()
+
+			if(!density)//no logging if we're open, because
+				log_game("[key_name(user)] has successfully opened a firelock with their bare hands \
+				[number_of_people_trying_to_open > 1 ? "along with [number_of_people_trying_to_open - 1] others" : ""], \
+				starting at [starting_health] health and ending at [user.health], their loc starting at [user_loc_starting_temperature] kelvin \
+				and ending at [users_air.temperature] kelvin.")
+
+		else if(user.stat != CONSCIOUS)
+			var/stat_string = "fallen into soft crit"
+			switch(user.stat)
+				if(UNCONSCIOUS)
+					stat_string = "fallen unconscious"
+				if(HARD_CRIT)
+					stat_string = "fallen into hard crit"
+				if(DEAD)
+					stat_string = "died"
+
+			log_game("[key_name(user)] has failed to open a firelock with their bare hands \
+			[length(people_trying_to_open) > 1 ? "along with [length(people_trying_to_open) - 1] others" : ""] because they have [stat_string], they started at [starting_health] health  \
+			they were trying to open the door for [(REALTIMEOFDAY - starting_time) / 10] seconds, their loc starting at [user_loc_starting_temperature] kelvin and ending at [users_air.temperature] kelvin.")
+
+			people_trying_to_open -= user
+
+		user.move_resist = initial(user.move_resist)
+
+/obj/machinery/door/firedoor/attack_paw(mob/living/user, list/modifiers)
+	. = ..()
+	if(!user.combat_mode)
+		attack_hand(user, modifiers)
+
+///deals burn damage to the user depending on whether theyre resistant to heat and how hot the door is, used as a callback in attack_hand's dynamic_do_after
+/obj/machinery/door/firedoor/process(delta_time)
+	for(var/mob/living/user as anything in people_trying_to_open)
+		if(QDELETED(user))
+			people_trying_to_open -= user
+			continue
+		//figure out how "hot" the door is with the temperature of the turf we are on and the user touching us, remember we dont conduct heat to the other side
+		var/datum/gas_mixture/our_air = return_air()
+		var/our_temperature = our_air.temperature
+		var/datum/gas_mixture/users_air = user.return_air()
+		var/users_temperature = users_air.temperature
+
+		var/heat_of_contact_surface = max(our_temperature, users_temperature)
+
+		if(heat_of_contact_surface < MINIMUM_TEMPERATURE_TO_BURN_ARMS)
+			return
+
+		var/heat_protected = FALSE
+		if(HAS_TRAIT(user, TRAIT_RESISTHEAT) || HAS_TRAIT(user, TRAIT_RESISTHEATHANDS))
+			heat_protected = TRUE
+
+		if(!iscarbon(user))
+			if(!heat_protected)
+				user.adjustFireLoss(5, forced = TRUE)
+			return
+
+		var/mob/living/carbon/carbon_user = user
+
+		if(carbon_user.gloves)
+			var/obj/item/clothing/gloves/gloves_of_user = carbon_user.gloves
+
+			if(gloves_of_user.max_heat_protection_temperature)
+				heat_protected ||= (gloves_of_user.max_heat_protection_temperature >= heat_of_contact_surface)
```

Comment:
I guess the linter doesn't realize this is valid yet. Change this to `heat_protected = heat_protected || (gloves_of_user.max_heat_protection_temperature >= heat_of_contact_surface)` then.